Active Documents/contenteditable-disabled.html

```diff
@@ -0,0 +1,192 @@
+        </p>
+        <pre class="example">
+            &lt;div contenteditable="true" contenteditable-disabled="
+                formatbackcolor
+                formatdent
+                formatfont
+                formatjustify
+                formatscript
+                formatstrikethrough
+                formatunderline
+                inserthorizontalrule">
+                This is a contenteditable region of a document that supports:
+                &lt;ul>
+                    &lt;li>bold text&lt;/li>
+                    &lt;li>italic text&lt;/li>
+                    &lt;li>links&lt;/li>
+                    &lt;li>lists&lt;/li>
+                    &lt;li>and possibly other formatting outside the scope of this specification.&lt;/li>
+                &lt;/ul>
+                
+                This contenteditable region explicitly does not support:
+                &lt;ul>
+                    &lt;li>background colors&lt;/li>
+                    &lt;li>indentation&lt;/li>
+                    &lt;li>font name, size or color customization&lt;/li>
+                    &lt;li>justification&lt;/li>
+                    &lt;li>superscript or subscript text&lt;/li>
+                    &lt;li>strikethrough&lt;/li>
+                    &lt;li>underlines&lt;/li>
+                    &lt;li>horizontal rules&lt;/li>
+                &lt;/ul>
+            &lt;/div>
+        </pre>
+    </section>
+    <section id="contenteditable-disabled">
+        <h2>
+            <code>contenteditable-disabled</code> DOM Content Attribute.
+        </h2>
+        <p>
+            The <dfn><code>contenteditable-disabled</code></dfn> attribute indicates which user agent-supplied editing UI SHOULD be disabled.  If specified, the attribute must have a value that is an <a href=https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#unordered-set-of-unique-space-separated-tokens>unordered
+            set of unique space-separated tokens</a> [[HTML]] which
+            are <a href=https://infra.spec.whatwg.org/#ascii-case-insensitive>ASCII case-insensitive</a> [[INFRA]].
+        </p>
+        <p>
+            The allowed set of tokens are listed below.  Each token matches the name of an <a href=https://www.w3.org/TR/input-events-1/#events-inputevents>Input Type</a> [[input-events-1]], or is a shortcut that is equivalent to specifying a set tokens.
+        </p>
+        <p>
+            When a token is specified, any user agent-supplied editing UI, that when invoked, would result in a beforeinput event having and inputType that matches one of the tokens in the attribute's value, or matches a token that is represented by a shortcut that is included in the attribute's value, SHOULD be disabled, omitted, or otherwise indicated by the user agent to be not applicable.  The exact mechanism for that indication is beyond the scope of this specification.
```

Comment:
The specification is kinda sad that it gives zero assurances about anything 😿 It's SHOULD everywhere, which leads me to believe something is not right with the overall approach.

Reply:
I believe the background for this is that it has been mentioned by browser developers that some of the appearance is determined by the OS and that it's not always possible to fully remove certain menus and menu entries. So in those cases they will grey those out in instead or make them visually disabled in some other way.

But using "SHOULD" everywhere also basically means that a browser can be fully compliant with the spec without actually having implemented anything at all, right? In that would it make more sense to write that it "MUST" disable menus in some way without specifying what that way is and those browsers that cannot comply with that then need to be listed as non-compliant?

Reply:
> So in those cases they will grey those out in instead or make them visually disabled in some other way.

Ok, that's interesting. You are saying the functionality get disabled... so, that could work: It's not so much the presentation/UI aspect that the spec needs to concern itself with, but with the functionality itself... then it can be "MUST" just around the functionality.  

> But using "SHOULD" everywhere also basically means that a browser can be fully compliant with the spec without actually having implemented anything at all, right?

Yeah. 

>  In that would it make more sense to write that it "MUST" disable menus in some way without specifying what that way is and those browsers that cannot comply with that then need to be listed as non-compliant?

So yeah, don't think about this as menus (because then it crosses into areas where browsers compete at the UI level)... just talk about functionality.

Reply:
We could say that user agents MUST NOT display UI that indicates an editing command is applicable if that editing command's corresponding Input Type is present in the value of the contenteditabledisabled attribute?